crates/diesel_utils/src/pagination.rs

```diff
@@ -0,0 +1,222 @@
+use crate::connection::DbPool;
+use base64::{Engine, prelude::BASE64_URL_SAFE_NO_PAD};
+use i_love_jesus::{PaginatedQueryBuilder, SortDirection};
+#[cfg(feature = "full")]
+use lemmy_utils::error::LemmyResult;
+use serde::{Deserialize, Serialize};
+use serde_with::skip_serializing_none;
+
+pub trait PaginationCursorBuilderNew {
```

Comment:
Please name this like `PaginationCursorBase64IdAndDirection`, or `PaginationCursorV2`.

Using `New` is a big no-no, especially since in another issue you're talking about a `NewNew`